### PR TITLE
Frontend linting fixes and action

### PR DESCRIPTION
**Pull request by sdevalapurkar**
# Overview

This PR includes the following proposed change(s):

- Frontend linting fixes and action
- Down to 0 errors, 14 warnings

## Type of change

- [x] Refactoring / Documentation
- [x] Technical debt

## How Has This Been Tested?

- Everything still seems to be working locally

### CODE REVIEW

**Comment by dogcomplex** [comment]
Wanted to pass on the good will and review this - looking good, simple changes but make sense (so far).  Can follow up when out of draft status 👍

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Linting Checks
+
+on: [pull_request]
+
+jobs:
+  build:
```
Comment (NickPhura) [inline .github/workflows/lint.yml:6]: Should add a name param below (will make the job title more readable in the actions view)

```yaml
  build:
    name: Lint App
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: "10"
+      - name: Install dependencies
+        run: npm install
+        working-directory: app
```
Comment (NickPhura) [inline .github/workflows/lint.yml:17]: Can we add either a step, or a job, that also lints the api and database folders

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Linting Checks
```
Comment (NickPhura) [inline .github/workflows/lint.yml:1]: I was actually playing around with a workflow for this yesterday, but never quite finished messing with it all.  So this is great!

File: .github/workflows/lint.yml (hunk 2 above)
Comment (sdevalapurkar) [reply .github/workflows/lint.yml:17]: I will do this later, as if I do it now the api and db lint will fail and we will have a failing action
I will work on lint fixes first then add the workflows to this respectively

File: .github/workflows/lint.yml (hunk 3 above)
Comment (sdevalapurkar) [reply .github/workflows/lint.yml:1]: yeah I would like this action to be a required check on pr merge too, that way we dont go backwards in terms of cleanly linted code :)

Comment (NickPhura) [reply .github/workflows/lint.yml:1]: yeah for sure

File: .github/workflows/lint.yml (hunk 2 above)
Comment (NickPhura) [reply .github/workflows/lint.yml:17]: Ah fair point

**Review by NickPhura — APPROVED** [review]

**Pull request merged**